samples/wifi_provisioning_over_ble/src/main.c

```diff
@@ -174,6 +174,7 @@ void do_wifi_connect(void)
 		printk("\r\nmalloc failed\r\n");
 		return;
 	}
+	memset(cnx_params, 0, sizeof(struct wifi_connect_req_params));
```

Comment:
Thank you to spotting the issue. Your patch fixes the issue. However, I suggest to clean up this piece of garbage a bit more.

There is in fact a memory leak, since `cnx_params` is never released. In fact, `net_mgmt()` does not require `cnx_params` to be allocated after the call (it would be an unusual API). Then, the code leverage assignment during init. So it can written:

    struct wifi_connect_req_params cnx_params = {
	.channel = WIFI_CHANNEL_ANY;
	.security = sec_type;
	.psk_length = strlen(pwd);
	.psk = pwd;
	.ssid_length = strlen(ssid);
	.ssid = ssid;
    };
    [...]